sql/catalyst/src/test/scala/org/apache/spark/sql/types/DecimalSuite.scala

```diff
@@ -94,8 +94,8 @@ class DecimalSuite extends SparkFunSuite with PrivateMethodTester {
     checkValues(Decimal(2e18.toLong), 2e18, 2e18.toLong)
     checkValues(Decimal(Long.MaxValue), Long.MaxValue.toDouble, Long.MaxValue)
     checkValues(Decimal(Long.MinValue), Long.MinValue.toDouble, Long.MinValue)
-    checkValues(Decimal(Double.MaxValue), Double.MaxValue, 0L)
-    checkValues(Decimal(Double.MinValue), Double.MinValue, 0L)
+    assert(Decimal(Double.MaxValue).toDouble == Double.MaxValue)
+    assert(Decimal(Double.MinValue).toDouble == Double.MinValue)
```

Comment:
why change this?

Reply:
because `toLong` overflows in this case

Reply:
I think we have reverted the changes in `Decimal.toLong`?

Reply:
yes, I can revert these changes and the tests pass, but here we are basically checking that we are returning wrong values, so I don't think it makes any sense. Anyway I can revert the change if you think we should.

Reply:
Yes we know `Decimal.toLong` can return wrong result, and we are going to fix it by making cast follow SQL standard. Let's revert it and fix it in the cast PR.